### ISSUE TITLE
Copy Paste Issues In JSON Mode

### ISSUE THREAD

**Issue by sureshbuzz** [ajaxorg/ace#2598]
Hi, 

I was using ace editor in multiple modes like editor.session.setMode("ace/mode/javascript"), editor.session.setMode("ace/mode/html"),editor.session.setMode("ace/mode/css"), editor.session.setMode("ace/mode/json")....etc. In Json mode i was facing copy paste issues. when cut the data operation is reflecting but data isnt getting copied to clipboard.  But in other modes it is working fine.

In "ace/keyboard/textinput" , if i make BROKEN_SETDATA=true rather than useragent.isChrome < 18 . It is working fine, I need to know is it right way to overcome this issue?

**Comment by nightwing** [comment]
Hi
Mode shouldn't have any effect on the clipboard, do you try to cut the same text in all modes?
Also which browser are you using?

setting BROKEN_SETDATA=true isn't a good way, since it will fall back to to setting texatarea value instead of doing clipboardData.setData, which is slower.

**Comment by sureshbuzz** [comment]
Yes. I tried to cut and paste in other modes, There it is working fine. I'm using google chrome. at oncopy declaration i was getting copied data . but while coming to data.insertion it is still reflecting old data.

**Comment by nightwing** [comment]
Could you please give me a testcase or url to a page where this happens. I am not able to reproduce this issue on http://plnkr.co/edit/g6vloG0n3bIGzOcez5IH?p=preview
